File: src/main/org/nlogo/prim/etc/_ifelsevalue.java
package org.nlogo.prim.etc;

import org.nlogo.api.LogoException;
import org.nlogo.nvm.Reporter;
import org.nlogo.nvm.Syntax;

public final strictfp class _ifelsevalue
    extends Reporter
    implements org.nlogo.nvm.Pure {
  @Override
  public Syntax syntax() {
    return Syntax.reporterSyntax
        (new int[]{Syntax.TYPE_BOOLEAN,
            Syntax.TYPE_REPORTER_BLOCK,
            Syntax.TYPE_REPORTER_BLOCK},
            Syntax.TYPE_WILDCARD);
  }

  @Override
  public Object report(final org.nlogo.nvm.Context context) throws LogoException {
    if (argEvalBooleanValue(context, 0)) {
      return args[1].report(context);
    } else {
      return args[2].report(context);
    }
  }
}

File: src/main/prim/etc/_ifelsevalue.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim.etc

import org.nlogo.api.Syntax
import org.nlogo.nvm.{ Context, Pure, Reporter }

class _ifelsevalue extends Reporter with Pure {
  override def syntax =
    Syntax.reporterSyntax(
      Array(Syntax.BooleanType,
            Syntax.ReporterBlockType,
            Syntax.ReporterBlockType),
      Syntax.WildcardType)
  override def report(context: Context): AnyRef =
    if (argEvalBooleanValue(context, 0))
      args(1).report(context)
    else
      args(2).report(context)
}
